walkthroughs/howto-mutual-tls-acm/infrastructure/deploy.yaml

```diff
@@ -0,0 +1,892 @@
+Description: >
+  This template deploys a VPC, with a pair of public and private subnets spread 
+  across two Availabilty Zones. It deploys an Internet Gateway, with a default 
+  route on the public subnets. It deploys a pair of NAT Gateways (one in each AZ), 
+  and default routes for them in the private subnets.
+
+Parameters:
+
+  EnvironmentName:
+    Description: An environment name that will be prefixed to resource names
+    Type: String
+
+  VpcCIDR: 
+    Description: Please enter the IP range (CIDR notation) for this VPC
+    Type: String
+    Default: 10.0.0.0/16
+
+  PublicSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the public subnet in the first Availability Zone
+    Type: String
+    Default: 10.0.0.0/19
+
+  PublicSubnet2CIDR:
+    Description: Please enter the IP range (CIDR notation) for the public subnet in the second Availability Zone
+    Type: String
+    Default: 10.0.32.0/19
+
+  PrivateSubnet1CIDR:
+    Description: Please enter the IP range (CIDR notation) for the private subnet in the first Availability Zone
+    Type: String
+    Default: 10.0.64.0/19
+
+  PrivateSubnet2CIDR:
+    Description: Please enter the IP range (CIDR notation) for the private subnet in the second Availability Zone
+    Type: String
+    Default: 10.0.96.0/19
+
+  LogGroupName:
+    Type: String
+    Default: "AppMeshExamples/mtls-acm-appmesh"
+
+  ECSServiceLogGroupRetentionInDays:
+    Type: Number
+    Default: 30
+
+  ECSServicesDomain:
+    Type: String
+    Description: "Domain name registered under Route-53 that will be used for Service Discovery"
+    Default: mtls-ec2.svc.cluster.local
+
+  TlsState:
+    Type: String
+    Default: mtls
+    Description: State of TLS in AWS App Mesh
+    AllowedValues:
+      - no-tls
+      - 1way-tls
+      - mtls
+
+  AppMeshMeshName:
+    Type: String
+    Description: Name of mesh
+
+  EnvoyImageName:
+    Type: String
+    Description: The image to use for the Envoy container
+
+  ColorTellerImageName:
+    Description: The name for the color teller image
+    Type: String
+
+  EC2Ami:
+    Description: EC2 AMI ID
+    Type: AWS::SSM::Parameter::Value<AWS::EC2::Image::Id>
+    Default: "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2"
+
+Conditions:
+  1wayTls: !Equals 
+    - !Ref TlsState
+    - 1way-tls
+  mtls: !Equals 
+    - !Ref TlsState
+    - mtls
+  setTls: !Or 
+    - Condition: mtls
+    - Condition: 1wayTls
+
+Resources:
+
+  VPC: 
```

Comment:
nit: We typically divide our CFN stacks into logical groups (network infrastructure, ECR repos, ECS cluster, mesh) so if one resource fails to create the rollback and create time won't be super long

Reply:
Sure. Based on other walkthroughs I can change this to:
- network infrastructure
- ACM
- ECR repos
- ECS cluster
- mesh
- ECS Service